### DISCUSSION TITLE
Enable scaling in workspaces

### DISCUSSION THREAD

**Discussion by timur-g** [linuxmint/discussions#67] [Ideas]
As long time Linux user i do not use workspaces - but scaling is exactly what I would need in them (to have 100%, 125%, 150%, 200%).. 
Not sure if scaling can be enabled in worskspaces, but if possible, please do.

**Comment by JosephMcc** [comment]
You mean allowing different display scaling per workspace. So like workspace1 is 100% and workspace2 is 125%?

**Reply by Kylemaul** [reply]
Exactly that! I'm not up to speed on the minutia of workspaces, but it would be a fantastic solution for many front end GUIs. (e.g. Steam, whose GUI seems to be tailored more for (smartphones?) than for modern fullscreen resolutions, lol) Ideally, it would also allow games, etc to be run in a more (native) resolution, as well.

**Reply by timur-g** [reply]
@JosephMcc Exactly. 
In my case, I test software in different resolutions and scalings and that would save me a lot of changing for my working workspace (when some apps go awry...).

**Reply by rcalixte** [reply]
@gitgt Have you considered [Distrobox](https://github.com/89luca89/distrobox) or something like the [Seamless Windows](https://docs.oracle.com/en/virtualization/virtualbox/6.0/user/seamlesswindows.html) feature for VirtualBox? Would something like this work for your use case?

**Reply by rcalixte** [reply]
Another possible viable workaround could be modifying Qt or GTK environment variables for scaling on the command-line or in the .desktop files. This would be bound to the application and not the workspace. Doing it via the command-line gives you more flexibility to iterate or tune rapidly while the .desktop files could be duplicated with each saved configuration if you use the same configuration repeatedly.

**Comment by timur-g** [comment]
@rcalixte Thank you for suggestions. I use VirtualBox from time to time but this would be too much, running other OS just to change resolution for an app that is already daily running in host OS. I will see Distrobox. Still, I think that initial request is the most simple.

**Reply by rcalixte** [reply]
It's simple to express maybe. Speaking for myself, coding something like scaling (fractional or otherwise) on a per-workspace basis would be considerably laborious and fraught with problems, whether it be X or Wayland. PRs are always welcome if you have ideas for how to implement it simply.

**Comment by Secret-chest** [comment]
Yes, per-app scaling makes more sense.

**Comment by timur-g** [comment]
I thought this would not be a bug thing, I think Gnome/GTK can be scaled globally with: 
gsettings set org.gnome.desktop.interface text-scaling-factor 2.0 
So, I was expecting GUI and something like: gsettings set org.gnome(cinnamon?).desktop.interface.workspace1 text-scaling-factor 2.0

**Reply by rcalixte** [reply]
The first option you mention exists in Cinnamon via System Settings -> Font Selection (or just Font Selection in the Main Menu). The latter option is unfortunately not that trivial.

Have you had a chance to look into the containerized/virtual solutions from before or the environment variable options proposed?

**Comment by timur-g** [comment]
I am looking into per-app scaling, if that could serve my need - thanks for suggestions. I still hope for this one in the longer run.

**Reply by rcalixte** [reply]
> I still hope for this one in the longer run.

To be clear, I don't believe there is a commercial GPU available today that can meet this ask. The oversampling to support fractional scaling (which is a global setting across all workspaces) has been known to put tremendous strain on GPUs. Trying to apply that on a per-workspace basis would require much more in resources. I would suggest reading into it further to understand the true nature of what it is you would be asking for.
